### PR TITLE
Support variables in configuration files

### PR DESCRIPTION
**Pull request by jmendeza**
https://github.com/craftercms/craftercms/issues/6624
Support variables in configuration files

### CODE REVIEW

File: src/main/java/org/craftercms/core/service/Context.java [hunk 1]
```diff
@@ -51,4 +53,10 @@ public interface Context extends Cloneable {
 
     Context clone();
 
+    /**
+     * Config variables for the context. e.g.: ${siteName} will be replaced by the value of the siteName variable
+     *
+     * @return Return a map of variables to be used when loading configuration files.
+     */
+    Map<String, String> getConfigLookupVariables();
```
Comment (jmendeza) [inline src/main/java/org/craftercms/core/service/Context.java:61]: Maybe we should call this something like attributes and make it more general

Comment (avasquez614) [reply src/main/java/org/craftercms/core/service/Context.java:61]: I'm ok with this name `configLookupVariables`

**Review by avasquez614 — APPROVED** [review]

File: src/test/java/org/craftercms/core/service/impl/ContentStoreServiceImplTest.java [hunk 2]
```diff
@@ -148,7 +148,7 @@ public void setUp() throws Exception {
         context = contentStoreService.getContext(null, FileSystemContentStoreAdapter.STORE_TYPE,
                                                  CLASSPATH_STORE_ROOT_FOLDER_PATH, DEFAULT_MERGING_ON,
                                                  DEFAULT_CACHE_ON, DEFAULT_MAX_ALLOWED_ITEMS_IN_CACHE,
-                                                 DEFAULT_IGNORE_HIDDEN_FILES);
+                                                 DEFAULT_IGNORE_HIDDEN_FILES, null);
```
Comment (sumerjabri) [inline src/test/java/org/craftercms/core/service/impl/ContentStoreServiceImplTest.java:151]: Shouldn't we add unit tests for this?

**Review by sumerjabri — CHANGES_REQUESTED** [review]

File: src/test/java/org/craftercms/core/service/impl/ContentStoreServiceImplTest.java (hunk 2 above)
Comment (jmendeza) [reply src/test/java/org/craftercms/core/service/impl/ContentStoreServiceImplTest.java]: Added a few test cases

**Review by sumerjabri — APPROVED** [review]

**Pull request merged**